### PR TITLE
chore(web-components): make web component tests async

### PR DESCRIPTION
**Pull request by radium-v**
## Previous Behavior

Some of the Playwright tests were still written in a serial style, which was originally adopted due to performance reasons when running Playwright tests on Github Actions with limited multithreading capabilities. This style is very brittle and causes a lot of flakiness for complex component tests.

Old style:
```ts
test.describe("Component", () => {
  let page: Page;
  let element: Locator;
  let root: Locator;

  test.beforeAll(async ({ browser }) => {
    page = await browser.newPage();
    element = page.locator('fluent-component');
    root = page.locator('#storybook-root');

    await page.goto('http://localhost:6006/iframe.html?storyId=component');
    await page.waitForFunction(() => customElements.whenDefined('fluent-component'));
  });

  test.afterAll(async () => {
    await page.close();
  });

  test("should render the component on the page", async () => {
    await root.evaluate((node) => {
      node.innerHTML = `<fluent-component>Hello</fluent-component>`;
    });

    await expect(element).toBeVisible();
  });
});
```

* Page and Locators are hoisted to the top, to be shared between components
* One page is used for ALL tests, requiring each subsequent test to overwrite the page state left by the previous test
* The use of `root` puts a hard dependency on the structure of the harness (currently Storybook's iframe page)
* If one test fails, all of the tests in the module have to be rerun

## New Behavior

This PR updates all remaining web component tests to use the standard style recommended by Playwright. See the [Playwright Documentation](https://playwright.dev/docs/writing-tests).

New style:
```ts
test.describe("Component", () => {
  test.beforeEach(async ({ page }) => {
    await page.goto('http://localhost:6006/iframe.html?storyId=component');
    await page.waitForFunction(() => customElements.whenDefined('fluent-component'));
  });

  test("should render the component on the page", async ({ page }) => {
    const element = page.locator('fluent-component');

    await page.setContent(`<fluent-component>Hello</fluent-component>`);

    await expect(element).toBeVisible();
  });
});
```

* Each test is run in its own browser page
* `page.setContent` is used to improve portability to other harnesses (less of a dependency on Storybook's `#storybook-root` element existing)
* Tests have their own state, so there's no cleanup required between tests

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: packages/web-components/public/100x100.png
Comment (radium-v) [inline packages/web-components/public/100x100.png:1]: I generated these with this script that I wrote a while back: https://gist.github.com/radium-v/1c038ef7a0c5e4bfd1758655047142f6

Using local images improves test reliability and security by not relying on external third-party sources in our test fixtures. While these services can be convenient for quick local prototyping, external assets can time-out and cause test failures if the service becomes unavailable or changes without our knowledge.

**Comment by fabricteam** [comment]
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

**Review by eljefe223 — APPROVED** [review]

File: change/@fluentui-web-components-3bfad2c2-151c-43c9-8f2f-c0218bc5f208.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-web-components-3bfad2c2-151c-43c9-8f2f-c0218bc5f208.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentui-web-components-v3 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32263&commitId=99ad56f1c25a4e68c695f87a1ced804078e14621&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar.Image.chromium.png|248| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

**Review by marchbox — APPROVED** [review]

**Review by mlijanto — APPROVED** [review]

**Pull request merged**